clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -38,6 +43,33 @@
     private ConnectionState connectionState = null;
     private HttpClient httpClient;
     private String stopError;
+    private Timer pingTimer = null;
+    private AtomicLong nextServerTimeout = new AtomicLong();
+    private AtomicLong nextPingActivation = new AtomicLong();
+    private Duration keepAliveInterval = Duration.ofSeconds(15);
+    private Duration serverTimeout = Duration.ofSeconds(30);
+    private Duration tickRate = Duration.ofSeconds(1);
+
+    public void setServerTimeout(Duration serverTimeout) {
```

Comment:
Are these things we want to make directly settable from the `HubConnection`? I guess we do it in the .NET client

Reply:
Yeah, they were put on the HubConnection because they can be set during the connection (in theory) and we didn't want to overload the initial options.